subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -95,13 +98,18 @@ static void internal_state_set(enum internal_state new_state)
 	internal_state = new_state;
 }
 
+static void set_current_job_id(uint8_t *job_id)
+{
+	strncpy(job_id_handling, job_id, sizeof(job_id_handling));
+	job_id_handling[sizeof(job_id_handling) - 1] = '\0';
```

Comment:
Since job IDs can be of arbitrary length between 1 and 64 bytes, I don't think any checks make sense here.
https://docs.aws.amazon.com/iot/latest/apireference/API_Job.html#iot-Type-Job-jobId

Reply:
I meant to check if the null-terminated byte overwrote the last character of the ID. Then we'd use the wrong ID in any subsequent operation. But i the absolute max length of an ID is 64 characters, then null-terminating at index 64 is always safe.